presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/Util.java

```diff
@@ -87,4 +88,54 @@ public static PlanNode transpose(PlanNode parent, PlanNode child)
                         node,
                         Assignments.identity(restrictedOutputs)));
     }
+
+    /**
+     * @return The original node, with identity projections possibly inserted between node and the child, limiting the columns to those permitted.
+     * Returns a present Optional iff the child was rewritten.
+     */
+    public static Optional<PlanNode> restrictChildOutputs(PlanNodeIdAllocator idAllocator, PlanNode node, Set<Symbol> permittedChildOutputs)
+    {
+        return restrictChildrenOutputs(idAllocator, node, ImmutableList.of(permittedChildOutputs));
+    }
+
+    /**
+     * @return The original node, with identity projections possibly inserted between node and each child, limiting the columns to those permitted.
+     * Returns a present Optional iff at least one child was rewritten.
+     */
+    public static Optional<PlanNode> restrictBinaryChildrenOutputs(
```

Comment:
I thought you wanted explicit handling of binary children, as opposed to lists, but I'll change it to be varargs.